### PR TITLE
#56 Canvas helper to add hw from iiif image info

### PR DESCRIPTION
**Pull request by rlskoeser**
implements `set_hwd_from_iiif` helper as described in #56 

I created a placeholder `set_hwd` for testing purposes; maybe this should not be merged until the other helper is implemented

- any other cases to handle with the image url passed in?
- wasn't sure what was desired for error handling; should we just document possible exceptions?
- I had to add `requests` to tox dependencies to get tests to work locally, shouldn't it pull from setup.py?

NOTE: this work depends on #70 . Once that is merged, this can be updated and the placeholder `set_hwd` method removed.

### CODE REVIEW

**Comment by digitaldogsbody** [comment]
> I created a placeholder `set_hwd` for testing purposes; maybe this should not be merged until the other helper is implemented

Agree, since the dummy implementation will need to be removed and it would be nicer to wait and remove that before merge.

>     * any other cases to handle with the image url passed in?
Not that I can immediately think of

>     * wasn't sure what was desired for error handling; should we just document possible exceptions?
I think that raising the errors to the main library is fine - we can document that the helper may raise and leave handling that up to the end user. I think this is better than silently failing and essentially requiring them to check for success after calling the function.

>     * I had to add `requests` to tox dependencies to get tests to work locally, shouldn't it pull from setup.py?
Afraid I'm not sure about this. @helrond, @p-galligan - any ideas?

**Comment by rlskoeser** [comment]
I'll set this to draft until the other helper has been merged

**Review by digitaldogsbody — APPROVED** [review]

File: iiif_prezi3/helpers/canvas_helpers.py [hunk 1]
```diff
@@ -3,21 +3,21 @@
...
+            anno_page_id (str): An HTTP URL for the annotation page to which the
+                annotation will be attached.
 
-    def add_image(self, image_url, anno_id=None, anno_page_id=None):
+        Returns:
+            anno_page (iiif-prezi3.skeleton.AnnotationPage): the AnnotationPage with
+                an Annotation and ResourceItem attached.
+        """
         body = ResourceItem(id=image_url, type='Image')
         annotation = Annotation(id=anno_id, body=body, target=image_url, motivation='painting', type='Annotation')
```
Comment (giacomomarchioro) [inline iiif_prezi3/helpers/canvas_helpers.py:22]: Just a comment, the type of the main IIIF object should be set by default from the JSON schema, so theoretically there should be no need for specifying it for Annotation and AnnotationPage, Canavses etc. etc.

**Review by giacomomarchioro — APPROVED** [review]
Impressive work, I think we should also think if it is the case of allowing to add directly a service using the `add_image` method.

**Comment by rlskoeser** [comment]
Earlier today the PR diff was showing a lot of changes I didn't make — I thought I had updated my fork and rebased my branch properly, not sure why it was flagging so many other changes. I just updated again and now the diff is accurate. Sorry about that.

**Comment by digitaldogsbody** [comment]
> Earlier today the PR diff was showing a lot of changes I didn't make — I thought I had updated my fork and rebased my branch properly, not sure why it was flagging so many other changes. I just updated again and now the diff is accurate. Sorry about that.

I think it merged rather than rebasing on top of the changes, because the commit list was showing a lot of commits from main branch.

Thanks for the great work - I'll merge based on the two approvals and follow the action instructions for getting the GH Pages setup working :)

**Pull request merged**